traits/tests/test_directory.py

```diff
@@ -52,6 +62,72 @@ def assign_invalid():
 
         self.assertRaises(TraitError, assign_invalid)
 
-    def test_fast(self):
-        example_model = FastExampleModel(path=gettempdir())
-        example_model.path = "."
+
+class TestBaseDirectory(unittest.TestCase):
+
+    def test_accepts_valid_dir_name(self):
+        foo = ExistsBaseDirectory()
+        tempdir = gettempdir()
+
+        self.assertIsInstance(tempdir, str)
+
+        foo.path = tempdir
+
+    def test_rejects_invalid_dir_name(self):
+        foo = ExistsBaseDirectory()
+
+        with self.assertRaises(TraitError):
+            foo.path = "!!!invalid_directory"
+
+    def test_rejects_valid_file_name(self):
+        foo = ExistsBaseDirectory()
+
+        with self.assertRaises(TraitError):
+            foo.path = __file__
+
+    @unittest.skipIf(sys.version_info < (3, 6), TESTS_SKIPPED_MESSAGE)
```

Comment:
Suggestion: you could reduce repetition and improve readability here by creating a single decorator at the top of the module. For example (untested):
```
requires_fspath = unittest.skipIf(..., "Test requires os.fspath, which is unavailable before Python 3.6")
```